vendor/knative.dev/pkg/webhook/certificates/resources/certs.go

```diff
@@ -47,11 +48,12 @@ func createCertTemplate(name, namespace string, notAfter time.Time) (*x509.Certi
 
 	serviceName := name + "." + namespace
 	commonName := serviceName + ".svc"
+	serviceHostname := network.GetServiceHostname(name, namespace)
 	serviceNames := []string{
 		name,
 		serviceName,
 		commonName,
-		serviceName + ".svc.cluster.local",
```

Comment:
well, yeah that ^ was a  mistake

Reply:
Why?  it was hoisted into a variable...

Reply:
the `svc.cluster.local` is a mistake, it will be wrong on some clusters. We have a method that parses the correct cluster local address